### ISSUE TITLE
[Week 3] Weekly LOs and Communication matters

### ISSUE THREAD

**Issue by okkhoy** [nusCS2113-AY1819S1/forum#7]
Dear all,

## Weekly LOs
Please note that the **deadline to achieve weekly LOs is the midnight before your tutorial**. 
We use scripts to detect your work; these scripts run at midnight and only the work that is done by midnight is eligible for marks (for the LOs that are graded).

## Communication matters
- You are encouraged to use this issue tracker to ask module related questions and doubts instead of directly shooting an email to your tutor.
- Reasons:
  - The discussions can benefit the whole class.
  - Additionally, it allows you to:
    1. post responses to announcements,
    2. communicate with the original poster for doubt clarification.

- Alternatively you can use the Slack channel too!
<br>

- You are encouraged to signup for notifications on both GitHub (please refer to #6 ) and Slack. 

Thanks,
Akshay

**Comment by QzSG** [comment]
Will we be informed in the case that LOs are not detected? Or if some are missing? Since it's the first time? 

With regards to the style guide we were supposed to follow 😱 , is there a reason why `case` inside `switch` statements aren't indented? Seems like a very rare style in particular for the switch formatting.

**Comment by cjunxiang** [comment]
Good day, 
have been asking around my class and none of us understood what LO, or what script is being mentioned. Would need some advice.

Thank you.

**Comment by QzSG** [comment]
> Good day,
> have been asking around my class and none of us understood what LO, or what script is being > mentioned. Would need some advice.
>
> Thank you.

@cjunxiang I think the only graded LO that is checked using the automated script is [[LO-1KLoC]](https://github.com/nusCS2113-AY1819S1/addressbook-level1/blob/master/docs/LearningOutcomes.adoc#work-in-a-1kloc-code-base-lo-1kloc) which is supposed to be done on [nusCS2113-AY1819S1/addressbook-level1](https://github.com/nusCS2113-AY1819S1/addressbook-level1) , it appears under `Week 3` -> `Evidence for LOs` under ` Can work with a 1KLoC code base `, the rest I assume just have to try out and show during tut

Perhaps @okkhoy can enlighten as some students already have had their tutorial this week or tomorrow

**Comment by cjunxiang** [comment]
I see! thanks @QzSG . Having a lot of trouble finding information on the website. Cheers. :-)

**Comment by IcedCoffeeBoy** [comment]
Hi,

How do we submit our work after we completed it? 
How do we distinguish between work that we do in own time and work that are needed to submit?  
@okkhoy

**Comment by okkhoy** [comment]
@QzSG Yes, the script will indicate* any missed LO as well. In general, if you have followed the instructions and used the naming specified in the instructions (note: naming conventions here is unrelated to coding standards and may not be applicable to all weeks!) the script should be able to detect your attempts.

WRT `case` being not indented: it is to avoid one level of unnecessary indentation. 

@cjunxiang This is a general announcement for **any** graded LO you attempt in this module. As @QzSG mentioned, for this week it is working on the 1kLoC code base of Addressbook-Level1.

We are trying to automate some of the things which were earlier done in the tutorials (thus spending precious 15-20 mins) on not-very-productive-work. These scripts are for this purpose.

@IcedCoffeeBoy Submission instructions are included in the LOs. For this week no special submission is required; as long as you pushed the changes to your fork of Addressbook-Level1 our scripts can detect them. 

-----

*The bot will be sending emails summarizing the achieved LO and missed LOs for a couple of weeks.
Once the team organization is set up, we will ask you to create a repository to receive summaries. Reason: We don't want to spam you when we have some SE tools that can be used for our purpose.

**Comment by Tsuweiquan** [comment]
Hello, 
I have done my LO (Fork, Clone then Enhance & Push) before 12am but did not receive any email/update on my LO. My friends have received some updates from the bot. 
Can i have some help with this?

**Comment by E0176882** [comment]
Hello,
I have done my LO (Fork, Clone then Enhance & Push) before 12am but the bot says I haven't updated as per the LO. My friends were able to have theirs detected.
Can i have some help with this?

**Comment by okkhoy** [comment]
@Tsuweiquan you haven't provided the correct GitHub username in the pre-module survey; the bot team did not have your records. Hence you don't have an email.
Please send me an **email** with the following details:

Full name:
Matric number:
NUSNET id:
NUSNET email:
GitHub username:

**Comment by okkhoy** [comment]
@E0176882 Please indicate when is your tutorial. We can check if the bot has some bug!

**Comment by E0176882** [comment]
Hi Sir, I'm @Tan Wei Hao, Joel<mailto:e0176882@u.nus.edu> not @Tsu Wei Quan<mailto:e0176844@u.nus.edu>.

The details are as follows:
Full name: Tan Wei Hao Joel
Matric Number: A0168449M
NUSNET id: E0176882
NUSNET email: E0176882@u.nus.edu
GitHub username: E0176882
________________________________

**Comment by Tsuweiquan** [comment]
@okkhoy I have emailed the details to you. Thank You.
